### ISSUE TITLE
possible bug in mesh blueprint verify failure reporting

### ISSUE THREAD

**Issue by cyrush** [LLNL/conduit#275]
While debugging a mesh, I noticed: 
```
      "errors": 
      [
        "mesh: reference to non-existent topologies 'topology'"
      ]
```

I began to suspect that "topology" was not the name of the topology, since this data set does not use `topology` to name its meshes.

Should this the quoted entry be `ref_name` instead of `field_name`?

```
//-----------------------------------------------------------------------------
bool verify_reference_field(const std::string &protocol,
                            const conduit::Node &node_tree,
                            conduit::Node &info_tree,
                            const conduit::Node &node,
                            conduit::Node &info,
                            const std::string &field_name,
                            const std::string &ref_path)
{
    bool res = verify_string_field(protocol, node, info, field_name);
    if(res)
    {
        const std::string ref_name = node[field_name].as_string();

        if(!node_tree.has_child(ref_path) || !node_tree[ref_path].has_child(ref_name))
        {
            log::error(info, protocol, "reference to non-existent " + ref_path +
                                        log::quote(field_name, 1));
            res = false;
        }
        else if(info_tree[ref_path][ref_name]["valid"].as_string() != "true")
        {
            log::error(info, protocol, "reference to invalid " + ref_path +
                                       log::quote(field_name, 1));
            res = false;
        }
    }

    log::validation(info[field_name], res);

    return res;
}
```

**Comment by cyrush** [comment]
@xjrc  can you take a look and let me know if you think this is a bug?

**Comment by xjrc** [comment]
@cyrush: I believe that you're right. The code should read:

```
//-----------------------------------------------------------------------------
bool verify_reference_field(const std::string &protocol,
                            const conduit::Node &node_tree,
                            conduit::Node &info_tree,
                            const conduit::Node &node,
                            conduit::Node &info,
                            const std::string &field_name,
                            const std::string &ref_path)
{
    bool res = verify_string_field(protocol, node, info, field_name);
    if(res)
    {
        const std::string ref_name = node[field_name].as_string();

        if(!node_tree.has_child(ref_path) || !node_tree[ref_path].has_child(ref_name))
        {
            log::error(info, protocol, "reference to non-existent " + field_name +
                                        log::quote(ref_name, 1));
            res = false;
        }
        else if(info_tree[ref_path][ref_name]["valid"].as_string() != "true")
        {
            log::error(info, protocol, "reference to invalid " + field_name +
                                       log::quote(ref_name, 1));
            res = false;
        }
    }

    log::validation(info[field_name], res);

    return res;
}
```

I'll submit a pull request with the fix.

**Comment by cyrush** [comment]
resolved in #276
